src/test/java/net/kaczmarzyk/spring/data/jpa/utils/SpecificationBuilderTest.java

```diff
@@ -68,8 +70,17 @@ public interface CustomSpecificationWithParam extends Specification<Customer> {
 	public interface CustomSpecificationWithHeader extends Specification<Customer> {
 	}
 
+	@BeforeEach
+	public void clearDb() {
+		customerRepo.deleteAll();
+	}
+
 	@Test
 	public void shouldCreateSpecificationDependingOnPathVar() {
+		Customer customer = customer("Homer", "Simpson")
```

Comment:
too little test data -- you have only 1 customer in db and then asserting that only 1 was found -- it would be true even if you did not applied filtering at all. add at least one more other customer (without pizza order)